utils/zero.go

```diff
@@ -0,0 +1,88 @@
+package utils
+
+import (
+	"reflect"
+)
+
+// IsZeroValue reports whether x is the zero value for its type.
+//
+// For basic types and slices of basic types, it uses a fast path without reflection.
+// For other types, it uses reflection.
+func IsZeroValue(x interface{}) bool {
+	// Fast path for basic types
+	switch v := x.(type) {
+	case nil:
+		return true
+	case bool:
+		return !v
+	case int:
+		return v == 0
+	case int8:
+		return v == 0
+	case int16:
+		return v == 0
+	case int32:
+		return v == 0
+	case int64:
+		return v == 0
+	case uint:
+		return v == 0
+	case uint8:
+		return v == 0
+	case uint16:
+		return v == 0
+	case uint32:
+		return v == 0
+	case uint64:
+		return v == 0
+	case uintptr:
+		return v == 0
+	case float32:
+		return v == 0.0
+	case float64:
+		return v == 0.0
+	case complex64:
+		return v == 0+0i
+	case complex128:
+		return v == 0+0i
+	case string:
+		return v == ""
+	case []bool:
+		return len(v) == 0
+	case []int:
+		return len(v) == 0
+	case []int8:
+		return len(v) == 0
+	case []int16:
+		return len(v) == 0
+	case []int32:
+		return len(v) == 0
+	case []int64:
+		return len(v) == 0
+	case []uint:
+		return len(v) == 0
+	case []uint8:
+		return len(v) == 0
+	case []uint16:
+		return len(v) == 0
+	case []uint32:
+		return len(v) == 0
+	case []uint64:
+		return len(v) == 0
+	case []uintptr:
+		return len(v) == 0
+	case []float32:
+		return len(v) == 0
+	case []float64:
+		return len(v) == 0
+	case []complex64:
+		return len(v) == 0
+	case []complex128:
+		return len(v) == 0
+	case []string:
+		return len(v) == 0
+	default:
+		// Slow path using reflection
+		return reflect.DeepEqual(x, reflect.Zero(reflect.TypeOf(x)).Interface())
```

Comment:
```go
type test string

func main() {
	x := []test{}
	fmt.Print(IsZeroValue(x))
}
```
will print false due to DeepEqual is pretty strict. Maybe this part should be adjusted?

Reply:
maybe this check is OK?
```go
val := reflect.ValueOf(x)
if val.Kind() == reflect.Slice && val.Len() == 0 {
	return true
}
```

Reply:
> maybe this check is OK above this part?
> 
> ```go
> val := reflect.ValueOf(x)
> if val.Kind() == reflect.Slice && val.Len() == 0 {
> 	return true
> }
> ```

Reply:
AFAIK in Go, a zero value for a slice is nil. An empty slice (like `x := []test{}` in your example) is not considered a zero value; it's just an empty slice.

Reply:
I already considered empty slices zero in the fast path. So that would have to be the same... I'll fix.

Reply:
> AFAIK in Go, a zero value for a slice is nil. An empty slice (like `x := []test{}` in your example) is not considered a zero value; it's just an empty slice.

Then we need to adjust switch-case part, right?

Reply:
Which behaviour would you consider correct?

Reply:
[cdb2885](https://github.com/gofiber/fiber/pull/2760/commits/cdb288542b077ecf5bb764d1d0033ba659c7a1dc) handles edge cases such as `x:= []test{}` and improves test/bench.

If we are going to consider those cases ZeroValues, otherwise we can revert to using DeepEqual.

Reply:
> > > AFAIK in Go, a zero value for a slice is nil. An empty slice (like `x := []test{}` in your example) is not considered a zero value; it's just an empty slice.
> > 
> > 
> > Then we need to adjust switch-case part, right?
> 
> Correct.
> 
> [cdb2885](https://github.com/gofiber/fiber/pull/2760/commits/cdb288542b077ecf5bb764d1d0033ba659c7a1dc) handles edge cases such as `x:= []test{}` and improves test/bench.
> 
> If we are going to consider those cases ZeroValues we can keep it, otherwise we can revert to using DeepEqual and remove the zero length slice cases.

That's OK to me but not sure `[]string{}` etc. is zero value according to Go. Should we either make switch case behavior same as reflect's zerovalue or make zero length slices zero value?

Reply:
https://github.com/gofiber/fiber/pull/2760/commits/c228dbd3c4c1fcc05c4508e568e31594bafa3446 aligns with conventions by removing slice cases and updating tests.